src/platform/i386/lapic.c

```diff
@@ -0,0 +1,175 @@
+#include "io.h"
+#include "kernel.h"
+#include "chal_cpu.h"
+#include "isr.h"
+
+#define APIC_DEFAULT_PHYS      0xfee00000
+#define APIC_HDR_LEN_OFF       0x04
+#define APIC_CNTRLR_ADDR_OFF   0x24
+
+#define LAPIC_SVI_REG          0x0F0
+#define LAPIC_EOI_REG          0x0B0
+#define LAPIC_TIMER_LVT_REG    0x320
+#define LAPIC_DIV_CONF_REG     0x3e0
+#define LAPIC_INIT_COUNT_REG   0x380
+#define LAPIC_CURR_COUNT_REG   0x390
+
+#define LAPIC_PERIODIC_MODE    (0x01 << 17)
+#define LAPIC_ONESHOT_MODE     (0x00 << 17)
+#define LAPIC_TSCDEADLINE_MODE (0x02 << 17)
+
+#define LAPIC_MASK (1<<15)
+
+#define IA32_MSR_TSC		0x00000010
+#define IA32_MSR_TSC_DEADLINE	0x000006e0
+
+#define TEST_DEADLINE 3400000
+#define RETRY_ITERS   3
+
+extern int timer_process(struct pt_regs *regs);
+
+enum lapic_timer_type {
+	ONESHOT = 0,
```

Comment:
These need to be properly namespaced.  These will conflict with other identifiers trivially.  Should be `LAPIC_ONESHOT` etc.